spec/lib/appsignal/probes/mri_spec.rb

```diff
@@ -75,29 +79,31 @@ def set_gauge(*args) # rubocop:disable Naming/AccessorMethodName
 
       it "tracks heap slots" do
         probe.call
-        expect_distribution_value("heap_slots", :heap_live)
-        expect_distribution_value("heap_slots", :heap_free)
+        expect_gauge_value("heap_slots", :tags => { :metric => :heap_live })
+        expect_gauge_value("heap_slots", :tags => { :metric => :heap_free })
       end
-    end
-  end
 
-  def expect_distribution_value(expected_key, metric, expected_value = nil)
-    expect(appsignal_mock.distribution_values).to satisfy do |distribution_values|
-      distribution_values.any? do |distribution_value|
-        key, value, metadata = distribution_value
-        next unless key == expected_key
-        next unless expected_value ? expected_value == value : !value.nil?
-        next unless metadata == { :metric => metric }
+      context "with custom hostname" do
+        let(:hostname) { "my hostname" }
 
-        true
+        it "reports custom hostname tag value" do
+          probe.call
+          expect_gauge_value("heap_slots", :tags => { :metric => :heap_live, :hostname => hostname })
+        end
       end
     end
   end
 
-  def expect_gauge_value(expected_key, expected_value = nil)
+  def expect_gauge_value(expected_key, expected_value = nil, tags: {})
+    expected_tags = { :hostname => Socket.gethostname }.merge(tags)
     expect(appsignal_mock.gauges).to satisfy do |gauges|
-      gauges.any? do |(key, value)|
-        expected_key == key && expected_value ? expected_value == value : !value.nil?
+      gauges.any? do |distribution_value|
+        key, value, tags = distribution_value
+        next unless key == expected_key
+        next unless expected_value ? expected_value == value : !value.nil?
+        next unless tags == expected_tags
```

Comment:
`next unless ...`? Shouldn't this be `return false unless ...`?

I suppose it doesn't make a difference -- next gives `nil`, which is falsy anyway. I've never seen it used this way before.

Reply:
If you call `return` the loop will exit and no other values will be checked. If you want it to exit out of that loop iteration with `false` you need to call `next false if ...` which is the same as `next if ...` for this because `nil` is falsy like you say.